proxy/proxy/common.go

```diff
@@ -222,6 +234,33 @@ func (c *ConnSet) Remove(id string, conn net.Conn) error {
 	return nil
 }
 
+// CloseIdle closes connections whose last activity exceeds the idle connection timeout.
+// This is used to drain connections during the shutdown period.
+func (c *ConnSet) CloseIdle(idleTimeout time.Duration, elapsedFn func(t time.Time) time.Duration) error {
```

Comment:
Why do we need `elapsedFn` here? Is there a situation when we won't want to use `time.Since`? (I see it's used in the tests - was that the motivation? If so, can we get away with testing with a very small time instead?)